internal/storage/bucket_handle_test.go

```diff
@@ -407,13 +408,99 @@ func (t *BucketHandleTest) TestListObjectMethodForMaxResult() {
 	AssertEq(TestObjectName, fourObj.Objects[3].Name)
 	AssertEq(nil, fourObj.CollapsedRuns)
 
+	// Validate that 2 objects are listed when MaxResults is passed 2.
 	AssertEq(nil, err2)
 	AssertEq(2, len(twoObj.Objects))
 	AssertEq(TestObjectRootFolderName, twoObj.Objects[0].Name)
 	AssertEq(TestObjectSubRootFolderName, twoObj.Objects[1].Name)
 	AssertEq(nil, twoObj.CollapsedRuns)
 }
 
+func (t *BucketHandleTest) TestListObjectMethodWithMissingMaxResult() {
+	// We have 4 objects in fakeserver.
+	fourObjWithMaxResults, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "",
+			Delimiter:                "",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "",
+			MaxResults:               100,
+			ProjectionVal:            0,
+		})
+
+	fourObjWithoutMaxResults, err2 := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "",
+			Delimiter:                "",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "",
+			//MaxResults:               2, // Missing MaxResults
+			ProjectionVal: 0,
+		})
+
+	// Validate that 4 objects are listed when MaxResults = 100, because there
+	// are total of 4 objects in fakebucket.
+	AssertEq(nil, err)
+	AssertEq(4, len(fourObjWithMaxResults.Objects))
```

Comment:
if this is validate the data before invoking the test please move it before test invocation.  why not just validate the data you received in previous call vs test invocation call directly.

Reply:
Removed the first invocation and validation block and directly validating the test results with the expected data.

Reply:
we need to check if there are 4 results before the test invocation.